test/api/api_test.go

```diff
@@ -189,37 +184,81 @@ func RunApiServer(t *testing.T, binary string, storageApiHost string) string {
 	return apiUrl
 }
 
+type ApiRequest struct {
+	Path   string      `json:"path"`
+	Method string      `json:"method"`
+	Body   interface{} `json:"body"`
+}
+
 // AssertExpectations compares expectations with the actual state.
 func AssertExpectations(
 	t *testing.T,
 	envProvider testhelper.EnvProvider,
 	testDirFs filesystem.Fs,
 	workingDirFs filesystem.Fs,
-	respCode int,
-	respBody string,
+	apiUrl string,
 	project *testproject.Project,
 ) {
 	t.Helper()
 
-	// Compare stdout
-	expectedResponseFile, err := testDirFs.ReadFile(filesystem.NewFileDef("expected-response"))
-	assert.NoError(t, err)
-	expectedRespBody := testhelper.ReplaceEnvsString(expectedResponseFile.Content, envProvider)
+	client := resty.New()
+	client.SetHostURL(apiUrl)
 
-	// Compare response status code
-	expectedCodeFile, err := testDirFs.ReadFile(filesystem.NewFileDef("expected-http-code"))
+	// request folders should be named e.g. 001-request1, 002-request2
+	dirs, err := testDirFs.Glob("[0-9][0-9][0-9]-*")
 	assert.NoError(t, err)
-	expectedCode := cast.ToInt(strings.TrimSpace(expectedCodeFile.Content))
-	assert.Equal(
-		t,
-		expectedCode,
-		respCode,
-		"Unexpected status code.\nRESPONSE:\n%s\n\n",
-		respBody,
-	)
-
-	// Assert response body
-	testhelper.AssertWildcards(t, expectedRespBody, respBody, "Unexpected response.")
+	for _, dir := range dirs {
+		// Read the request file
+		requestFile, err := testDirFs.ReadFile(filesystem.NewFileDef(dir + "/request.json"))
```

Comment:
`dir + "/request.json"` -> `filesystem.Join`.

Reply:
V tomto pripade by ti to slo, lebo vo virtualFs, pouzivame iba `/`, no napr s `os.ReadFile` by ti to na windows neslo.
Preto je best practice spaja vzdy `paths` cez `Join` funkciu.

Reply:
🙈 omg, vím, to už jsem dokonce i používal